Tests/DeckOfPlayingCardsTests/DeckTest.swift

```diff
@@ -22,41 +24,46 @@ class DeckTests: XCTestCase {
             countByPlayingCard[playingCard, default: 0] += 1
         }
 
-        XCTAssertEqual(countByPlayingCard.count, 52)
-        XCTAssertTrue(countByPlayingCard.values.allSatisfy { $0 == 1 })
+        #expect(countByPlayingCard.count == 52)
+        #expect(countByPlayingCard.values.allSatisfy( { $0 == 1 } ))
 
         for rank in Rank.allCases {
             for suit in Suit.allCases {
                 let playingCard = PlayingCard(rank: rank, suit: suit)
-                XCTAssertEqual(countByPlayingCard[playingCard], 1)
+                #expect(countByPlayingCard[playingCard] == 1)
             }
         }
     }
     
+    @Test
     func testDeal() {
         let playingCard = PlayingCard(rank: .ace, suit: .clubs)
         var deck: Deck = [playingCard]
         
-        XCTAssertEqual(deck.deal(), playingCard)
-        XCTAssertNil(deck.deal())
+        #expect(deck.deal() == playingCard)
+        #expect(deck.deal() == nil)
     }
     
+    @Test
     func testCountEmptyDeckHasZeroCards() {
         let deck = Deck()
-        XCTAssertEqual(deck.count, 0)
+        //XCTAssertEqual(deck.count, 0)
+        #expect(deck.count == 0)
     }
     
+    @Test
     func testCountStandard52CardDeckHas52Cards() {
         let deck = Deck.standard52CardDeck()
         
-        XCTAssertEqual(deck.count, 52)
+        #expect(deck.count == 52)
     }
     
-    func testCountDealingDecreasesCountByOne() {
+    @Test
+    func testCountDealingDecreasesCountByOne() throws {
         var deck = Deck([PlayingCard(rank: .ace, suit: .spades), PlayingCard(rank: .queen, suit: .hearts)])
         
-        XCTAssertEqual(deck.count, 2)
-        XCTAssertNotNil(deck.deal())
-        XCTAssertEqual(deck.count, 1)
+        #expect(deck.count == 2)
+        try #require(deck.deal() != nil)
```

Comment:
I like that you're using `#require`, and I'd go a step further and recommend taking the opportunity to inspect its return value. E.g.
```suggestion
        let dealtCard = try #require(deck.deal())
        #expect(/* validate something about `dealtCard` */)
```

Reply:
I just tried that, and I thought it would work seamlessly, but I'm getting an error about it not being able to be used on a mutating member:

<img width="1112" height="192" alt="Screenshot 2025-08-08 at 11 32 10 AM" src="https://github.com/user-attachments/assets/74883616-8fb4-4d24-9e9b-27c8cfa358e4" />

Since this was returning an optional and I wanted to "force it" to stop if that optional wasn't value - I was leaning into using `#require`, Is that too far or not aligned with it's intent?

Is there something I need to do or add to support using #require with a mutating function?

Reply:
Ah, no—I didn't realize it was mutating and the macros can't currently support such functions I'm afraid. @grynspan do you have any workarounds to suggest for suppressing our rich expansion (which are still somewhat idiomatic/non-ugly)? If not, we can drop this suggestion